baremetal/metal3machine_manager.go

```diff
@@ -52,11 +52,12 @@ const (
 	ProviderName = "baremetal"
```

Comment:
should the provider name reflect the overall naming change?
Meaning:
ProviderName = "metal3"

Reply:
This needs to be tackled in a separate PR. I will look into it